Utilities/DataLoading/EdFi.LoadTools/ApiClient/FileContextProvider.cs

```diff
@@ -0,0 +1,152 @@
+                {
+                    _log.Debug($"{contextPrefix} XSD validation is enabled.");
+                    xmlReaderSettings.Schemas = _xmlSchemaSet;
+                    xmlReaderSettings.ValidationType = ValidationType.Schema;
+
+                    xmlReaderSettings.ValidationEventHandler += (s, e) => { _log.Error($"{e.Message}"); };
+                }
+
+                using (var stream = new FileStream(fileName, FileMode.Open, FileAccess.Read))
+                {
+                    if (stream.Length == 0)
+                    {
+                        _log.Warn($"{contextPrefix} Empty file.");
+                        return fileContext;
+                    }
+
+                    using (var reader = XmlReader.Create(stream, xmlReaderSettings))
+                    {
+                        // get the interchange name
+                        if (reader.MoveToContent() == XmlNodeType.Element)
+                        {
+                            var match = _interchangeNameRegex.Match(reader.Name);
+
+                            if (match.Success)
+                            {
+                                fileContext.FileType = match.Groups["InterchangeType"].Value;
+                            }
+                            else
+                            {
+                                _log.Warn($"{contextPrefix} Unknown file type");
+                                return fileContext;
+                            }
+                        }
+
+                        _log.Debug($"{contextPrefix} Checking for reference identities");
+
+                        // loop through and count the ids refs
+                        while (reader.Read())
+                        {
+                            if (reader.NodeType != XmlNodeType.Element)
+                            {
+                                continue;
+                            }
+
+                            string refId = reader.GetAttribute("ref");
+
+                            if (!string.IsNullOrEmpty(refId))
+                            {
+                                fileContext.NumberOfIdRefs++;
+                            }
+                        }
+
+                        fileContext.IsValid = true;
+                    }
+                }
+
+                if (fileContext.NumberOfIdRefs > 0)
+                {
+                    _log.Debug($"{contextPrefix} Reference identities exist");
+                }
+
+                _log.Debug($"{contextPrefix} Finished processing.");
+
+                if (_xsdConfiguration.DoNotValidateXml)
+                {
+                    return fileContext;
+                }
+
+                if (fileContext.IsValid)
+                {
+                    _log.Info($"{contextPrefix} passed XSD validation.");
+                }
+                else
+                {
+                    _log.Warn($"{contextPrefix} failed XSD validation.");
```

Comment:
Might be helpful to mention that this file will be skipped?

Reply:
added a debug statement